airflow-core/src/airflow/ui/src/components/DateTimeInput.tsx

```diff
@@ -32,8 +32,11 @@ type Props = {
 export const DateTimeInput = forwardRef<HTMLInputElement, Props>(({ onChange, value, ...rest }, ref) => {
   const { selectedTimezone } = useTimezone();
 
-  // Make the value timezone-aware
-  const date = dayjs(value).tz(selectedTimezone).format("YYYY-MM-DDTHH:mm:ss.SSS");
+  // Make the value timezone-aware, handle invalid dates
+  const date =
+    value !== "" && dayjs(value).isValid()
```

Comment:
`isValid()` would handle `value === ""` case. Leaving one condition is enough and clearer IMO.
```suggestion
    dayjs(value).isValid()
```

Reply:
+1

Reply:
SURE! Makes perfect sense!